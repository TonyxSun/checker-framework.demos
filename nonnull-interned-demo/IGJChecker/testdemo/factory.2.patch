Index: src/checkers/types/AnnotatedTypeFactory.java
===================================================================
--- src/checkers/types/AnnotatedTypeFactory.java	(revision 3580)
+++ src/checkers/types/AnnotatedTypeFactory.java	(working copy)
@@ -481,10 +481,10 @@
         if (elt == null)
             throw new IllegalArgumentException("null element");
         if (elementCache.containsKey(elt))
-            return elementCache.get(elt);
+            return atypes.deepCopy(elementCache.get(elt));
         AnnotatedTypeMirror type = fromElement(elt);
         annotateImplicit(elt, type);
-        elementCache.put(elt, type);
+        elementCache.put(elt, atypes.deepCopy(type));
         return type;
     }
 
