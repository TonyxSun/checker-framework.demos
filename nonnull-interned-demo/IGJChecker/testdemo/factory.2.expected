
BUILD SUCCESSFUL
Total time: 4 seconds
